ui/tests/acceptance/setup.old.js

```diff
@@ -0,0 +1,121 @@
+import { setDefaultTimeout, After, Before, defineParameterType } from 'cucumber'
+import { createSession, closeSession, client, startWebDriver, stopWebDriver } from 'nightwatch-api'
+import { rollbackConfigs, setConfigs, cacheConfigs } from './helpers/config'
+import { getAllLogsWithDateTime } from './helpers/browserConsole.js'
+const codify = require('./helpers/codify')
+
+const ldap = require('./helpers/ldapHelper')
+
+const RUNNING_ON_CI = !!process.env.CI
+const RUNNING_ON_SAUCELABS = !!process.env.SAUCE_USERNAME
+
+const CUCUMBER_LOCAL_TIMEOUT = 60000
+const CUCUMBER_DRONE_TIMEOUT = 180000
+const SAUCELABS_ASYNC_SCRIPT_TIMEOUT = 10000
+const CUCUMBER_TIMEOUT = RUNNING_ON_CI ? CUCUMBER_DRONE_TIMEOUT : CUCUMBER_LOCAL_TIMEOUT
+setDefaultTimeout(CUCUMBER_TIMEOUT)
+
+let env = RUNNING_ON_CI ? 'drone' : 'local'
+env = RUNNING_ON_SAUCELABS ? 'saucelabs' : env
+
+defineParameterType({
+  name: 'code',
+  regexp: /"([^"\\]*(\\.[^"\\]*)*)"|'([^'\\]*(\\.[^'\\]*)*)'/,
+  type: String,
+  transformer: s => codify.replaceInlineCode(s)
+})
+
+Before(function startDriverOnLocal() {
+  return RUNNING_ON_CI || startWebDriver({ env })
+})
+
+Before(function createSessionForEnv() {
+  return createSession({ env })
+})
+
+Before(function logSessionInfoOnSauceLabs() {
+  if (process.env.SAUCE_USERNAME) {
+    return client
+      .session(function(session) {
+        console.log('  Link to saucelabs job: https://app.saucelabs.com/tests/' + session.sessionId)
+      })
+      .timeoutsAsyncScript(SAUCELABS_ASYNC_SCRIPT_TIMEOUT)
+  }
+})
+
+Before(function createLdapClient() {
+  if (client.globals.ocis) {
+    return ldap.createClient().then(ldapClient => {
+      client.globals.ldapClient = ldapClient
+    })
+  }
+})
+
+After(function deleteLdapClient() {
+  if (client.globals.ocis && client.globals.ldapClient) {
+    return ldap.terminate(client.globals.ldapClient)
+  }
+})
+
+async function cacheAndSetConfigs(server) {
+  if (client.globals.ocis) {
+    return
+  }
+  await cacheConfigs(server)
+  return setConfigs(server, client.globals.backend_admin_username)
+}
+
+Before(function cacheAndSetConfigsOnLocal() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A function should not mix return statements with and without a result.](https://app.codacy.com/gh/owncloud/ocis-hello/pullRequest?prid=5709816)